include/zephyr/kernel.h

```diff
@@ -3416,6 +3416,97 @@ static inline unsigned int z_impl_k_sem_count_get(struct k_sem *sem)
 
 /** @} */
 
+#if defined(CONFIG_SCHED_IPI_SUPPORTED) || defined(__DOXYGEN__)
+struct k_ipi_work;
```

Comment:
I was sure I had just commented on this but I may not have it sent or something. This would need to be documented too ("something something opaque structure representing an IPI work item") but since you must be pretty tired with my comments by now it's fine to keep for later (many kernel objects are lacking such documentation actually IIRC, so maybe someone will tackle them all at once :))